scripts/migration/migrate-addon.sh

```diff
@@ -0,0 +1,653 @@
+  "ConfigMap/promsd-sidecar/istio-system"
+  "Service/istio-operator.istio-operator"
+  "Service/istio-citadel/istio-system"
+  "Service/istio-galley/istio-system"
+  "Service/istio-pilot/istio-system"
+  "Service/istio-policy/istio-system"
+  "Service/istio-sidecar-injector/istio-system"
+  "Service/istio-telemetry/istio-system"
+  "Service/promsd/istio-system"
+  "Deployment/istio-operator/istio-operator"
+  "Deployment/istio-citadel/istio-system"
+  "Deployment/istio-galley/istio-system"
+  "Deployment/istio-ingressgateway/istio-system"
+  "Deployment/istio-pilot/istio-system"
+  "Deployment/istio-policy/istio-system"
+  "Deployment/istio-sidecar-injector/istio-system"
+  "Deployment/istio-telemetry/istio-system"
+  "Deployment/promsd/istio-system"
+  "HorizontalPodAutoscaler/istio-ingressgateway/istio-system"
+  "HorizontalPodAutoscaler/istio-pilot/istio-system"
+  "HorizontalPodAutoscaler/istio-policy/istio-system"
+  "HorizontalPodAutoscaler/istio-telemetry/istio-system"
+  "attributemanifest/istioproxy/istio-system"
+  "attributemanifest/kubernetes/istio-system"
+  "handler/kubernetesenv/istio-system"
+  "handler/stackdriver-mixer/istio-system"
+  "instance/attributes/istio-system"
+  "instance/mixer-request-count/istio-system"
+  "instance/mixer-request-latency/istio-system"
+  "rule/kubeattrgenrulerule/istio-system"
+  "rule/stackdriver-mixer/istio-system"
+  "rule/tcpkubeattrgenrulerule/istio-system"
+  "DestinationRule/istio-policy/istio-system"
+  "DestinationRule/istio-telemetry/istio-system"
+  "Secret/cacerts/istio-system"
+  "Secret/istio-ca-secret/istio-system"
+)
+
+die() {
+    echo -e "${red}$*${clr}" >&2 ; exit 1;
+}
+
+kube() {
+  echo -e "${grey}Running: kubectl $*${clr}"  >&2
+  kubectl --context="${CONTEXT}" "$@"
+}
+
+heading() {
+  echo -e "\n${blue}${1} ${clr}"
+}
+
+prompt() {
+  if [[ -n "${SKIP_CONFIRM}" ]]; then
+    heading "${1}"
+    return
+  fi
+  echo -en "\n${blue}${1} ${clr}"
+  read -r -p "Continue? [Y/n] " response
+  case "$response" in
+      "") return ;;
+      [yY][eE][sS]|[yY]) return ;;
+  esac
+  exit 1
+}
+
+prompt_unsupported() {
+  echo -e "${yellow}${1}${clr}\n${2}"
+  if [[ -n "${SKIP_CONFIRM}" ]]; then
+    die "--skip-confirm set but an unsupported configuration was detected"
+  fi
+  read -r -p "Continue anyways? [y/N] " response
+  case "$response" in
+      [yY][eE][sS]|[yY]) return ;;
+  esac
+  exit 1
+}
+
+print_intro() {
+  if [[ -z "${SKIP_CONFIRM}" ]]; then
+    echo
+    echo "This tool automatically upgrades the Istio addon in the current cluster from 1.4 to Anthos Service Mesh."
+    echo "Selected cluster: '$(kubectl config current-context 2> /dev/null)'"
+    echo
+    prompt "Starting migration..."
+  fi
+}
+
+check_prerequisites() {
+  echo -e "${blue}Checking prerequisites...${clr}"
+  command -v kubectl >| /dev/null 2>&1 || die "kubectl must be installed, aborting."
+  command -v istioctl >| /dev/null 2>&1 || die "istioctl must be installed, aborting."
```

Comment:
I don't think I've ever once installed `istioctl`. Can we just have this tool download and extract it if necessary?

Reply:
The tool isn't meant to run in isolation. Since we also expect the Managed control-plane to be installed prior to the migration tool, that step should end up installing istioctl. The docs will reflect this.

Reply:
We can rely on the docs to guide users to download `istioctl`, but "installing" it (e.g. putting it in $PATH) isn't something that's normally done (to my knowledge). We should support a way to either specify the `istioctl` binary location or put it in a consistent location ourselves.

Reply:
To be more specific: The first step in the docs is to use the scriptaro installer to download the appropriate version of istioctl. The location of the binary is one of the arguments to the installer script. We could add the same argument here to this script and add logic to find istioctl there - does that work?

Reply:
Sure. Or we could make it `./istioctl` and have users `cd` into the output directory that `install_asm` makes, since it puts a copy of `istioctl` there.